### ISSUE TITLE
xmlprotection filters out DTD but still makes request specified in DTD

### ISSUE THREAD

**Issue by bjornhilstad** [membrane/service-proxy#346]
We are using membrane 4.6.2 to proxy some soap services.
We use <xmlProtection> and see in debug log that DTDs are filtered out as expected.
But we also see that membrane still tries to get the resource specified in the DTD which could open us up for XML External Entity attack (https://www.owasp.org/index.php/XML_External_Entity_(XXE)_Processing)

Example config:
```
<serviceProxy port="8095">			
  <path>/somepath</path>            			
  <xmlProtection/>
  <target host="localhost" port="8088"/>
</serviceProxy>

```
Example request:
```
<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE data SYSTEM "http://localhost:8089/some.dtd">
<soapenv:Envelope .....

```

This would result in the DTD being removed from the request that hits localhost:8088, but a request to get some.dtd from localhost:8089 would be made.

Is this something that could be avoided?

**Comment by predic8** [comment]
Thanks for reporting that and the detailed explanation.
We will have a look at this and fix it.

**Comment by xorpherion** [comment]
Hello @bjornhilstad 
could you try out the master branch and see if your problems are resolved? Thanks!

**Comment by bjornhilstad** [comment]
Yes, this change seems to resolve the issue. Built from master and could not see requests for the DTD anymore.
Thanks for a quick resolution and an otherwise excellent product.
Can we expect that this fix will go into 4.7.2?

**Comment by rrayst** [comment]
Yes, 4.7.2 was cut from master after the fix.
